core/templates/dev/head/pages/exploration_editor/translation_tab/StateTranslationDirective.js

```diff
@@ -0,0 +1,145 @@
+            if (tabId === $scope.TAB_ID_FEEDBACK) {
+              if (!$scope.stateDefaultOutcome || !$scope.stateInteractionId) {
+                return true;
+              } else {
+                return false;
+              }
+            } else if (tabId === $scope.TAB_ID_HINTS) {
+              if ($scope.stateHints.length <= 0) {
+                return true;
+              } else {
+                return false;
+              }
+            } else {
+              if (!$scope.stateSolution) {
+                return true;
+              } else {
+                return false;
+              }
+            }
+          };
+
+          $scope.changeActiveHintIndex = function(newIndex) {
+            // If the current hint is being clicked on again, close it.
+            if (newIndex === $scope.activeHintIndex) {
+              $scope.activeHintIndex = null;
+            } else {
+              $scope.activeHintIndex = newIndex;
+            }
+          };
+
+          $scope.changeActiveAnswerGroupIndex = function(newIndex) {
+            // If the current answer group is being clicked on again, close it.
+            if (newIndex === $scope.activeAnswerGroupIndex) {
+              $scope.activeAnswerGroupIndex = null;
+            } else {
+              $scope.activeAnswerGroupIndex = newIndex;
+            }
+          };
+
+
+          $scope.getHtmlSummary = function(subtitledHtml) {
+            var htmlAsPlainText = $filter(
+              'formatRtePreview')(subtitledHtml.getHtml());
+            return htmlAsPlainText;
+          };
+
+          $scope.$on('refreshStateTranslation', function() {
+            $scope.initStateTranslation();
+          });
+
+          $scope.initStateTranslation = function() {
+            $scope.activatedTabId = $scope.TAB_ID_CONTENT;
+
+            if (!EditorStateService.getActiveStateName()) {
+              EditorStateService.setActiveStateName(
+                ExplorationInitStateNameService.displayed);
+            }
+
+            var stateName = EditorStateService.getActiveStateName();
+
+            $scope.stateContent = ExplorationStatesService
+              .getStateContentMemento(stateName);
+            $scope.stateSolution = ExplorationStatesService
+              .getSolutionMemento(stateName);
+            $scope.stateHints = ExplorationStatesService
+              .getHintsMemento(stateName);
+            $scope.stateAnswerGroups = ExplorationStatesService
+              .getInteractionAnswerGroupsMemento(stateName);
+            $scope.stateDefaultOutcome = ExplorationStatesService
+              .getInteractionDefaultOutcomeMemento(stateName);
+            $scope.stateInteractionId = ExplorationStatesService
+              .getInteractionIdMemento(stateName);
+            $scope.activeHintIndex = null;
+            $scope.activeAnswerGroupIndex = null;
+            $rootScope.loadingMessage = '';
+          };
+          if (ExplorationStatesService.isInitialized()) {
```

Comment:
Can you add a TODO comment here? Even though this technically works, it's not a very clean solution and is a bit of a hack. And at the least, a developer looking at this will likely have no idea what's going on here so we should give them a heads up. I think the comment could be "We need to call initStateTranslation() here in case the listener that receives 'refreshStateTranslation' is not set by the time it is broadcasted from ExplorationEditor.js. Figure out a solution that does not rely on covering the race condition."

Reply:
Done!